tests/test_spectrum_processing.py

```diff
@@ -0,0 +1,100 @@
+import numpy as np
+from matchms import Spectrum
+from ms2query.spectrum_processing import require_peaks_below_mz, \
+    spectrum_processing_minimal, spectrum_processing_s2v
+
+
+
+def test_require_peaks_below_mz_no_params():
+    """Test default"""
+    mz = np.array([110, 220, 330, 440], dtype="float")
+    intensities = np.array([0, 1, 10, 100], dtype="float")
+    spectrum_in = Spectrum(mz=mz, intensities=intensities)
+
+    spectrum = require_peaks_below_mz(spectrum_in)
+
+    assert spectrum is None, "Expected None because the number of peaks (4) is less than the default threshold (10)."
+
+
+def test_require_peaks_below_mz_required_4():
+    """Test with adjustment of n_required."""
+    mz = np.array([110, 220, 330, 440], dtype="float")
+    intensities = np.array([0, 1, 10, 100], dtype="float")
+    spectrum_in = Spectrum(mz=mz, intensities=intensities)
+
+    spectrum = require_peaks_below_mz(spectrum_in, n_required=4)
+
+    assert spectrum == spectrum_in, "Expected the spectrum to qualify because the number of peaks (4) is equal to the" \
+                                "required number (4)."
+
+
+def test_require_peaks_below_mz_required_4_below_max_mz():
+    """Test with adjustment of n_required but with lower max_mz."""
+    mz = np.array([110, 220, 330, 440], dtype="float")
+    intensities = np.array([0, 1, 10, 100], dtype="float")
+    spectrum_in = Spectrum(mz=mz, intensities=intensities)
+
+    spectrum = require_peaks_below_mz(spectrum_in, n_required=4, max_mz=400)
+
+    assert spectrum is None, \
+        "Expected None since peaks with mz<=max_mz is less than n_required."
+
+
+def test_require_peaks_below_empty_spectrum():
+    spectrum_in = None
+    spectrum = require_peaks_below_mz(spectrum_in)
+
+    assert spectrum is None, "Expected different handling of None spectrum."
+
+
+def test_spectrum_processing_minimal_default():
```

Comment:
I am not really sure what the correct way of testing a function that only calls other functions is, since all functions used already have their own test functions. One way I could think of is running one test for each function used (but not all tests for each function), to make sure they are all called. In that case a few test functions would be missing, but the current solution of testing one positive and one negative case might already be good enough.

Reply:
That is precisely why I added `sonarcloud`. I gives a very nice overview of the test coverage of our code. If you click on the code coverage link there it will highlight all uncovered lines. That also includes if a function is run from another function, so it is not strictly necessary to test every function individually. I usually do that when I want to test something very specific or aim for coverage of some conditions which are otherwise missing.

Reply:
Oh, and would often also just decide pragmatically. One test too much (e.g. already covered elsewhere) is not too bad as long as this doesn't become excessive. And sonarcloud will then point us to missing aspects that are not tested yet.